tests/test_port_dpb_vlan.py

```diff
@@ -0,0 +1,235 @@
+import time
+import pytest
+from port_dpb import Port
+from port_dpb import DPB
+
+@pytest.mark.usefixtures('dpb_setup_fixture')
+@pytest.mark.usefixtures('dvs_vlan_manager')
+class TestPortDPBVlan(object):
+    def check_syslog(self, dvs, marker, log, expected_cnt):
+        (exitcode, num) = dvs.runcmd(['sh', '-c', "awk \'/%s/,ENDFILE {print;}\' /var/log/syslog | grep \"%s\" | wc -l" % (marker, log)])
+        assert num.strip() >= str(expected_cnt)
+
+    '''
+    @pytest.mark.skip()
+    '''
+    def test_dependency(self, dvs):
+
+        vlan = "100"
+        p = Port(dvs, "Ethernet0")
+        p.sync_from_config_db()
+        self.dvs_vlan.create_vlan(vlan)
+        #print "Created VLAN100"
+        self.dvs_vlan.create_vlan_member(vlan, p.get_name())
+        #print "Added Ethernet0 to VLAN100"
+        marker = dvs.add_log_marker()
+        p.delete_from_config_db()
+        #Verify that we are looping on dependency
+        time.sleep(2)
+        self.check_syslog(dvs, marker, "Cannot remove port as bridge port OID is present", 1)
+        assert(p.exists_in_asic_db() == True)
+
+        self.dvs_vlan.remove_vlan_member(vlan, p.get_name())
+        time.sleep(1)
+        # Verify that port is deleted
+        assert(p.exists_in_asic_db() == False)
+
+        #Create the port back and delete the VLAN
+        p.write_to_config_db()
+        #print "Added port:%s to config DB"%p.get_name()
+        p.verify_config_db()
+        #print "Config DB verification passed!"
+        p.verify_app_db()
+        #print "Application DB verification passed!"
+        p.verify_asic_db()
+        #print "ASIC DB verification passed!"
+
+        self.dvs_vlan.remove_vlan(vlan)
```

Comment:
Would recommend verifying that there's no VLAN at the end of this test case so we can guarantee that the next test case starts in a clean state.

Reply:
This goes for the rest of the test cases as well.

Reply:
👍